geomesa-raster/src/main/scala/org/locationtech/geomesa/raster/util/RasterUtils.scala

```diff
@@ -0,0 +1,291 @@
+    if (image.getWidth == newWidth && image.getHeight == newHeight) {
+      image
+    } else {
+      if (newWidth < 1 || newHeight < 1) null
+      else {
+        val result = allocateBufferedImage(newWidth, newHeight, image)
+        val resGraphics = result.createGraphics()
+        resGraphics.setRenderingHint(RenderingHints.KEY_INTERPOLATION, RenderingHints.VALUE_INTERPOLATION_NEAREST_NEIGHBOR)
+        resGraphics.drawImage(image, 0, 0, newWidth, newHeight, null)
+        resGraphics.dispose()
+        result
+      }
+    }
+  }
+
+  def cropRaster(raster: Raster, cropEnv: Envelope): Option[BufferedImage] = {
+    val rasterEnv = raster.referencedEnvelope
+    val intersection = rasterEnv.intersection(envelopeToReferencedEnvelope(cropEnv))
+    if (intersection.equals(rasterEnv)) {
+      // If the intersection of the crop envelope and tile is equal to the tile envelope we are done.
+      Some(renderedImageToBufferedImage(raster.chunk))
+    } else {
+      // Check the area of the intersection to ensure it is at least 1x1 pixels
+      val chunkWidth   = raster.chunk.getWidth
+      val chunkHeight  = raster.chunk.getHeight
+      val chunkXRes    = rasterEnv.getWidth / chunkWidth
+      val chunkYRes    = rasterEnv.getHeight / chunkHeight
+      //TODO: check for corner cases: https://geomesa.atlassian.net/browse/GEOMESA-758
+      val widthPixels  = Math.round(intersection.getWidth / chunkXRes)
+      val heightPixels = Math.round(intersection.getHeight / chunkYRes)
+      if (widthPixels > 0 && heightPixels > 0) {
+        // Now that we know the area is at least 1x1 perform the cropping operation
+        val uLX = Math.max(Math.floor((intersection.getMinX - rasterEnv.getMinimum(0)) / chunkXRes).toInt, 0)
+        val uLY = Math.max(Math.floor((rasterEnv.getMaximum(1) - intersection.getMaxY) / chunkYRes).toInt, 0)
+        val tempWidth     = Math.max(Math.ceil(intersection.getWidth / chunkXRes).toInt, 0)
+        val finalWidth    = if (tempWidth + uLX > chunkWidth) chunkWidth - uLX else tempWidth
+        val tempHeight    = Math.max(Math.ceil(intersection.getHeight / chunkYRes).toInt, 0)
+        val finalHeight   = if (tempHeight + uLY > chunkHeight) chunkHeight - uLY else tempHeight
+        val bufferedChunk = renderedImageToBufferedImage(raster.chunk)
+        Some(bufferedChunk.getSubimage(uLX, uLY, finalWidth, finalHeight))
+      } else None
+    }
+  }
+
+  def envelopeToReferencedEnvelope(e: Envelope): ReferencedEnvelope = {
+    new ReferencedEnvelope(e.getMinimum(0),
+      e.getMaximum(0),
+      e.getMinimum(1),
+      e.getMaximum(1),
+      DefaultGeographicCRS.WGS84)
+  }
+
+  def getNewImage[T: TypeTag](w: Int, h: Int, fill: Array[T],
+                              imageType: Int = BufferedImage.TYPE_BYTE_GRAY): BufferedImage = {
+    val image = new BufferedImage(w, h, imageType)
+    val wr = image.getRaster
+    val setPixel: (Int, Int) => Unit = typeOf[T] match {
+      case t if t =:= typeOf[Int]    =>
```

Comment:
Okay, I suspected I wasn't going to win this point so easily. Does it bomb at runtime, during compile, or does IDEA complain?